Mathlib/MeasureTheory/Group/ModularCharacter.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2025 Noam Atar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Noam Atar
+-/
+import Mathlib.MeasureTheory.Function.LocallyIntegrable
+import Mathlib.MeasureTheory.Group.Integral
+import Mathlib.MeasureTheory.Group.Measure
+import Mathlib.Topology.Metrizable.Urysohn
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+import Mathlib.MeasureTheory.Constructions.BorelSpace.Basic
+
+/-!
+# Modular character of a locally compact group
+
+On a locally compact group, there is a natural homomorphism `G → ℝ≥0*`, which for `g : G` gives the
+value `μ (· * g⁻¹) / μ`, where `μ` is an (inner regular) Haar measure. This file defines this
+homomorphism, called the modular character, and shows that it is independent of the chosen Haar
+measure.
+
+TODO: Show that the character is continuous.
+
+## Main Declarations
+
+* `modularCharacterFun`: Define the modular character, which is `g ↦ haarScalarFactor
+  (map (· * g) ν) ν`, where ν is the Haar measure given by (the noncomputable)
+  `MeasureTheory.Measure.haar`. The result that this does not depend on the measure chosen is
+  `modularCharacterFun_eq_haarScalarFactor`.
```

Comment:
What you're describing here is the technical implementation of the function, but that's not really relevant for a user. Something like: "the amount by which right multiplication by `g` rescales the Haar measure" (or should it be `g⁻¹`?) is more informative. Same thing elsewhere in this file, above and below.